mon-pix/app/components/routes/campaigns/assessment/skill-review.js

```diff
@@ -119,23 +119,22 @@ export default class SkillReview extends Component {
   }
 
   @action
-  shareCampaignParticipation() {
+  async shareCampaignParticipation() {
     this.displayErrorMessage = false;
-    this.displayLoadingButton = true;
-    const campaignParticipation = this.args.model.campaignParticipation;
-    return campaignParticipation.save()
-      .then(async () => {
-        campaignParticipation.isShared = true;
-
-        const isSimplifiedAccessCampaign = await this.args.model.campaignParticipation.campaign.get('isSimplifiedAccess');
-        if (isSimplifiedAccessCampaign) {
-          return this.disconnectUser();
-        }
-      })
-      .catch(() => {
-        campaignParticipation.rollbackAttributes();
-        this.displayErrorMessage = true;
-      });
+    const campaignParticipationResult = this.args.model.campaignParticipationResult;
+
+    const adapter = this.store.adapterFor('campaign-participation-result');
+    try {
+      await adapter.share(campaignParticipationResult.id);
+    } catch (err) {
+      this.displayErrorMessage = true;
+    }
+
+    campaignParticipationResult.isShared = true;
+    const isSimplifiedAccessCampaign = this.args.model.campaign.isSimplifiedAccess;
+    if (isSimplifiedAccessCampaign) {
+      return this.disconnectUser();
+    }
```

Comment:
Comme c'est écrit, s'il y a une erreur au moment du partage, le code de la ligne 133 à 136 sera exécuté et la compagne sera marquée comme partagée: `campaignParticipationResult=isShared`

On devrait plutôt écrire:
```js
    const adapter = this.store.adapterFor('campaign-participation-result');
    try {
      await adapter.share(campaignParticipationResult.id);

      campaignParticipationResult.isShared = true;
      const isSimplifiedAccessCampaign = this.args.model.campaign.isSimplifiedAccess;
      if (isSimplifiedAccessCampaign) {
        return this.disconnectUser();
      }
    } catch (err) {
      this.displayErrorMessage = true;
    }
```

Reply:
Peut-être ajouter un test pour vérifier que `campaignParticipationResult.isShared = false` quand on a une erreur lors du partage ?

Reply:
Avec ta proposition, du coup si l'erreur se produit sur la déconnexion du user, on affiche le message d'erreur qui n'est pas du coup associé à ce type d'erreur donc ça ne va pas non plus.

Peut être qu'on peut `return this.displayErrorMessage = true` :
```suggestion
    const adapter = this.store.adapterFor('campaign-participation-result');
    try {
      await adapter.share(campaignParticipationResult.id);
    } catch (err) {
      return this.displayErrorMessage = true;
    }

    campaignParticipationResult.isShared = true;
    const isSimplifiedAccessCampaign = this.args.model.campaign.isSimplifiedAccess;
    if (isSimplifiedAccessCampaign) {
      return this.disconnectUser();
    }
```

Reply:
Je trouve pas ça top de faire un return sur une assignation. Du coup, je propose:

```
this.displayErrorMessage = true;
return;
```